Add k shortest paths (#1763)
igraph/igraph: CHANGELOG.md

 - `igraph_roots_for_tree_layout()` computes a set of roots suitable for a nice tree layout.
 - `igraph_fundamental_cycles()` computes a fundamental cycle basis (experimental).
 - `igraph_minimum_cycle_basis()` computes an unweighted minimum cycle basis (experimental).
 - `igraph_strvector_merge()` moves all strings from one string vectors to the end of another without re-allocating them.

>  - `igraph_get_k_shortest_paths()` to find at most k alternative shortest paths between a source and a target vertex (#1763, thanks to @GroteGnoom)

Reviewer: This does not find `k` "alternative" shortestest paths, but simple _the `k` shortest paths_. They are not alternatives. They are the shortest, the 2nd shortest, the 3rd shortest, and so on.

Reviewer: ```suggestion
 - `igraph_get_k_shortest_paths()` finds the k shortest paths between a source and a target vertex (#1763, thanks to @GroteGnoom)
```

Reviewer: ```suggestion
 - `igraph_get_k_shortest_paths()` finds the k shortest paths between a source and a target vertex (#1763, thanks to @GroteGnoom)
```
